win: test: cleanup temporary resources on exit

janlazo opened the pull request:
Extracted from #8405 because these tests don't depend on the build script.

janlazo committed:
win: test: disable non-admin failing tests

mkfifo (msysgit) does not work outside of msys2 environment.
gzip tests fail on Windows.

mklink requires admin privs for file symbolic links so mklink fails.

janlazo committed:
win: test: don't test symlink if not admin user

janlazo committed:
win: test: delete sautest/

janlazo committed:
win: test: close shada file before os.remove

justinmk commented on test/functional/legacy/delete_spec.lua:
```diff
@@ -52,6 +52,10 @@ describe('Test for delete()', function()
       endif
     ]])
+    if eval('v:shell_error') == 1 then
+      eq(0, eval("delete('Xfile')"))
```
Probably should just do `os.remove('Xfile')` in `before_each`, since this isn't a useful test.

justinmk commented on test/functional/legacy/delete_spec.lua:
```diff
@@ -52,6 +52,10 @@ describe('Test for delete()', function()
         silent !ln -s Xfile Xlink
       endif
     ]])
+    if eval('v:shell_error') == 1 then
```
I've read that exit codes are not reliable on Windows, so maybe `eval('v:shell_error') != 0` is better practice.

janlazo committed:
fixup: always delete Xfile, fix exit code check

after_each + os.remove ensures Xfile is deleted after every test.
Windows exit code is inconsistent.
